coral/src/app/features/connectors/request/ConnectorPromotionRequest.tsx

```diff
@@ -0,0 +1,364 @@
+          <_Controller<ConnectorRequestFormSchema>
+            name={"connectorConfig"}
+            control={form.control}
+            render={({ field, fieldState: { error } }) => {
+              return (
+                <>
+                  <BorderBox
+                    borderColor="grey-20"
+                    borderWidth={1}
+                    paddingY={"3"}
+                    borderRadius={2}
+                  >
+                    <MonacoEditor
+                      data-testid="connector-config"
+                      height="200px"
+                      language="json"
+                      theme={"light"}
+                      onChange={(value) => {
+                        value !== undefined &&
+                          form.setValue("connectorConfig", value, {
+                            shouldValidate: true,
+                          });
+                      }}
+                      options={{
+                        language: "json",
+                        ariaLabel: "Connector configuration, required",
+                        renderControlCharacters: false,
+                        minimap: { enabled: false },
+                        folding: false,
+                        lineNumbers: "off",
+                        scrollBeyondLastLine: false,
+                        renderLineHighlight: "none",
+                        cursorBlinking: "solid",
+                        overviewRulerLanes: 0,
+                        wordBasedSuggestions: false,
+                        lineNumbersMinChars: 3,
+                        glyphMargin: false,
+                        cursorStyle: "line-thin",
+                        scrollbar: {
+                          useShadows: false,
+                          verticalScrollbarSize: 2,
+                        },
+                        fixedOverflowWidgets: true,
+                      }}
+                      value={field.value}
+                    />
+                  </BorderBox>
+                  <Box marginTop={"1"} marginBottom={"3"} aria-hidden={"true"}>
+                    <Typography.Caption color={"error-50"}>
+                      {error !== undefined ? error.message : <>&nbsp;</>}
+                    </Typography.Caption>
+                  </Box>
+                </>
+              );
+            }}
+          />
+
+          <Grid colGap={"l1"} marginTop={"3"} cols={"2"}>
+            <Textarea<ConnectorRequestFormSchema>
+              name={"description"}
+              value={connectorDetails?.connectorContents?.description}
+              labelText={"Connector description (read-only)"}
+              readOnly={true}
+            />
+            <Textarea<ConnectorRequestFormSchema>
+              name={"remarks"}
+              labelText={"Message for approval"}
+            />
+          </Grid>
+
+          <Box display={"flex"} colGap={"l1"} marginTop={"3"}>
+            <SubmitButton loading={isLoadingRequestPromotion}>
```

Comment:
Ah, great point! I found out how to make it work. But I would make a separate PR for that, since I also want to update tests etc in `Form` and didn't want to make this PR too big.

Reply:
There are many way to make it work, haha! But indeed needs its own PR ^^